### ISSUE TITLE
Is it possible to add "grad_scale" argument in the lsoftmax operator?

### ISSUE THREAD

**Issue by nicklhy** [luoyetx/mx-lsoftmax#3]
I guess grad_scale can be important for each loss function. Thus it would be better if you can add this argument in your operator.

**Comment by luoyetx** [comment]
@nicklhy `grad_scale` should be added in a loss operator, but lsoftmax operator is not a loss operator. It acts like a fully connected operator. If you mean weight the fully connected output and lsoftmax output, you can set `beta`.

**Comment by nicklhy** [comment]
@luoyetx , I checked your code again and understood it. Thank you for your reply!
BTW, have you evaluated this method on some larger datasets(i.e. ImageNet)?

**Comment by luoyetx** [comment]
@nicklhy It works on larger dataset. I have tested on CASIA-Webface and MS-1M-Celeb.

**Comment by nicklhy** [comment]
Nice work. Maybe you can submit a pr to merge your code to the official MXNet.

**Comment by luoyetx** [comment]
You can easily use this operator by copying 3 files to `mxnet/src/operator`.

**Comment by nttstar** [comment]
@luoyetx Could you give me some hints on training webface dataset? My softmax loss keep growing at the beginning.    params:  beta=1000, beta_min=5, margin=4, scale=0.99.   batch_size=128 on single GPU.

**Comment by nttstar** [comment]
It is going to work. I set scale=0.9997.
